cli/module_loader.rs

```diff
@@ -1123,80 +1219,134 @@ impl<TGraphContainer: ModuleGraphContainer>
+          _ => {
+            if self
+              .shared
+              .cjs_tracker
+              .is_cjs_with_known_is_script(specifier, *media_type, *is_script)
+              .map_err(JsErrorBox::from_err)?
+            {
+              return Ok(Some(CodeOrDeferredEmit::Cjs {
+                specifier,
+                media_type: *media_type,
+                source: &source.text,
+              }));
+            }
+            let code: ModuleCodeString = match media_type {
+              MediaType::JavaScript
+              | MediaType::Unknown
+              | MediaType::Mjs
+              | MediaType::Json => source.text.clone().into(),
+              MediaType::Dts | MediaType::Dcts | MediaType::Dmts => {
+                Default::default()
+              }
+              MediaType::Cjs | MediaType::Cts => {
+                return Ok(Some(CodeOrDeferredEmit::Cjs {
+                  specifier,
+                  media_type: *media_type,
+                  source: &source.text,
+                }));
+              }
+              MediaType::TypeScript
+              | MediaType::Mts
+              | MediaType::Jsx
+              | MediaType::Tsx => {
+                return Ok(Some(CodeOrDeferredEmit::DeferredEmit {
+                  specifier,
+                  media_type: *media_type,
+                  source: &source.text,
+                }));
+              }
+              MediaType::Css
+              | MediaType::Html
+              | MediaType::Sql
+              | MediaType::Wasm
+              | MediaType::SourceMap => {
+                panic!("Unexpected media type {media_type} for {specifier}")
+              }
+            };
+
+            // at this point, we no longer need the parsed source in memory, so free it
+            self.parsed_source_cache.free(specifier);
+
+            Ok(Some(CodeOrDeferredEmit::Code(ModuleCodeStringSource {
+              code: ModuleSourceCode::String(code),
+              found_url: specifier.clone(),
+              module_type: module_type_from_media_type(*media_type),
+            })))
           }
-        };
-
-        // at this point, we no longer need the parsed source in memory, so free it
-        self.parsed_source_cache.free(specifier);
-
-        Ok(Some(CodeOrDeferredEmit::Code(ModuleCodeStringSource {
-          code: ModuleSourceCode::String(code),
-          found_url: specifier.clone(),
-          media_type: *media_type,
-        })))
+        }
       }
       Some(deno_graph::Module::Wasm(WasmModule {
         source, specifier, ..
       })) => Ok(Some(CodeOrDeferredEmit::Code(ModuleCodeStringSource {
         code: ModuleSourceCode::Bytes(source.clone().into()),
         found_url: specifier.clone(),
-        media_type: MediaType::Wasm,
+        module_type: match requested_module_type {
+          RequestedModuleType::Bytes => ModuleType::Bytes,
+          RequestedModuleType::Text => ModuleType::Text, // nonsensical, but allowed
```

Comment:
Why is this nonsensical?

Reply:
Why would someone import a Wasm module as text?

Reply:
See line 1193. This is someone doing:

```ts
import text from "./math.wasm" with { type: "text" };
```